unix/src/machine_stats/__init__.py

```diff
@@ -283,6 +283,61 @@ def v2_playbook_on_stats(self, stats):
         self._display.display("", screen_only=True, stderr=True)
 
 
+class MeasurementsResultCallback(ResultCallback):
+    def v2_playbook_on_stats(self, stats):
+        """How to measure fields
+
+        The fields that need to be tracked can be added in the fields_to_measure list.
+        If it's a custom field, please add it to the custom_fields_to_measure list.
+        """
+
+        fields_to_measure = ["ram_used_gb"]
+        custom_fields_to_measure = ["cpu_average", "cpu_peak"]
+
+        """Process JSON payload
+
+        Go through each server in the results, and for the fields mentioned in the 
+        `fields_to_measure` or `custom_fields_to_measure`, add its measurements to the
+        transformed data.
+        """
+
+        transformed_json_payload = []
+        for server in list(self._total_results.values()):
+            for field in server:
+                # Add data (ram_used_gb) from fields_to_measure list to the transformed_json_payload list
+                if field in fields_to_measure:
+                    server_dict = {}
+                    server_dict["measurable_type"] = "server"
+                    server_dict["field_name"] = field + "_timeseries"
```

Comment:
Great! Makes sense adding the `cpu_utilization` field to the `custom_fields_to_measure` list. I'll add that change.

> we don't have such a thing as `ram_used_gb_timeseries`.

I am getting confused here between `cpu_*` and `ram_*` fields. In the machine-stats JSON, we have the `ram_used_gb` field regardless of any `cpu` related custom fields. Do you mind elaborating on this again?

Reply:
Sure thing, no problem. 

Full disclosure, I could be wrong about `ram_use_gb_timeseries`. we could dive a bit deeper and understand the reason behind having a **ram_use_gb_timeseries**.  That being said, we don't use it. 

Our efforts have been focused on CPU utilization (not on ram utilization). 

The way I understand it is that `_timeseries` is only associated with a "point in time" capture. That means, using the -`-cpu-utilization-only-value`,  since it will allow us to have individual data points rather than averages/aggregates.  

that is why I don't think we are using (or will use) `_timeseries` with `peak` and `average`.


The beauty of your implementation is that we can make this adjustment really easily by adding or removing values from the `fields` or `custom_fields` lists.

My suggestion is that we adjust your implementation to comply with the flow that we are currently targeting (CPU utilization  => --cpu-utilization-only-value). 

Once we get this one in we can figure out what we are going to do with `ram*` and `cpu` (peak and average).

I hope that helps, Please let me know